buildSrc/src/main/kotlin/com/datadog/gradle/plugin/jsonschema/PokoDeserializerGenerator.kt

```diff
@@ -483,7 +544,17 @@ class PokoDeserializerGenerator(
         return funBuilder.build()
     }
 
-    // endregion
+    private fun TypeDefinition.Class.isConstantClass(): Boolean {
+        // all the properties are of type Constant and the additionalProperties is null
+        this.properties.forEach {
+            if (it.type !is TypeDefinition.Constant) {
+                return false
+            }
+        }
+        return this.additionalProperties == null
```

Comment:
nitpick: You could use 
```kotlin
    return (this.additionalProperties == null) && this.properties.all { it.type is TypeDefinition.Constant }
```

Reply:
The reason I did it this way is for performances reasons. The `loop` will return false whenever that condition is not met so it will skip the other properties.

Reply:
The implementation of `all` also exits on the first element that fails ;)